base/regex.jl

```diff
@@ -152,8 +152,10 @@ function match(re::Regex, str::UTF8String, idx::Integer, add_opts::UInt32=UInt32
     RegexMatch(mat, cap, ovec[1]+1, off, re)
 end
 
+_utf8(str) = utf8(str)
```

Comment:
Why shouldn't it work on the following: `SubString(x,5)` where `typeof(x) -> UTF16String`?
Shouldn't it work for any `AbstractString`?
I'm right now in the process of adding tests for cases of `AbstractString`, where they are `SubString` of `UTF16String` and `UTF32String`.